### PR TITLE
[SYCLomatic] Use FWD_STRIDES/BWD_STRIDES to replace the deprecated INPUT_STRIDES/OUTPUT_STRIDES in fft_utils.hpp

### PR DESCRIPTION
**Pull request by zhiweij1**
Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

**Review by tomflinda — APPROVED** [review]
LGTM

File: clang/runtime/dpct-rt/include/dpct/fft_utils.hpp [hunk 1]
```diff
@@ -1059,279 +1061,208 @@ class fft_engine {
...
-      if (direction != _direction || is_this_compute_inplace != _is_inplace) {
-        if constexpr (Precision == oneapi::mkl::dft::precision::SINGLE) {
-          if (direction != _direction) {
-            swap_distance(_desc_sc);
-            _direction = direction;
-          }
-          if (is_this_compute_inplace != _is_inplace) {
-            _is_inplace = is_this_compute_inplace;
+      if ((is_this_compute_inplace != _is_inplace) ||
+          (is_complex && (direction != _direction))) {
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/dpct/fft_utils.hpp]: remove this if.
Use a flag to save state if goes into  1187 or 1195

File: clang/runtime/dpct-rt/include/dpct/fft_utils.hpp [hunk 2]
```diff
@@ -1059,279 +1061,208 @@ class fft_engine {
...
+          } else {
+            desc->set_value(oneapi::mkl::dft::config_param::PLACEMENT,
+                            oneapi::mkl::dft::config_value::NOT_INPLACE);
+            if (_is_basic && !is_complex)
+              set_stride_and_distance_basic<false>(desc);
           }
-          _desc_dc->commit(*_q);
+#endif
         }
+        desc->commit(*_q);
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/dpct/fft_utils.hpp]:
```suggestion
        if (need_commit)
          desc->commit(*_q);
```

File: clang/runtime/dpct-rt/include/dpct/fft_utils.hpp [hunk 3]
```diff
@@ -1059,279 +1061,208 @@ class fft_engine {
...
-    }                                                                          \
-  }
-
-  template <class T, oneapi::mkl::dft::precision Precision>
-  void compute_complex(T *input, T *output, fft_direction direction) {
+  template <class Dest_t, class T>
+  void compute_impl(Dest_t desc, T *input, T *output,
+                    std::optional<fft_direction> direction) {
     bool is_this_compute_inplace = input == output;
+    bool is_complex = direction.has_value();
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/dpct/fft_utils.hpp]:
```suggestion
    constexpr bool is_complex = Dest_t::element_type::domain == oneapi::mkl::dft::domain::COMPLEX;
```

**Review by ziranzha — APPROVED** [review]

**Pull request merged**